### PR TITLE
increase scope of a few classes to write APIs against them

### PR DESCRIPTION
**Pull request by mxm**
This pull request is similar to #8 in the sense that it exposes some internal API classes and fields. The intention is to write a custom `coGroupByKey` translator. For that, we need access to `CoGbkResultSchema`, `TaggedKeyedPCollection`, and `RawUnionValue`.

### CODE REVIEW

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/join/KeyedPCollectionTuple.java [hunk 1]
```diff
@@ -151,16 +151,25 @@ public void finishSpecifying() {
...
-  static class TaggedKeyedPCollection<K, V> {
+  public static class TaggedKeyedPCollection<K, V> {
+
     final TupleTag<V> tupleTag;
     final PCollection<KV<K, V>> pCollection;
 
     public TaggedKeyedPCollection(
-        TupleTag<V> tupleTag,
-        PCollection<KV<K, V>> pCollection) {
+            TupleTag<V> tupleTag,
```
Comment (davorbonaci) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/join/KeyedPCollectionTuple.java]: Please revert spacing changes, here and in the next row.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/join/KeyedPCollectionTuple.java [hunk 2]
```diff
@@ -151,16 +151,25 @@ public void finishSpecifying() {
...
+            PCollection<KV<K, V>> pCollection) {
       this.tupleTag = tupleTag;
       this.pCollection = pCollection;
     }
+
+    public PCollection<KV<K, V>> getCollection() {
```
Comment (davorbonaci) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/join/KeyedPCollectionTuple.java]: Could you please add Javadoc for getCollection() and getTupleTag()?

**Comment by davorbonaci** [comment]
Hi mxm,
Thanks for this contribution. We'd be happy to merge it.

Could you please address two very minor comments posted above?

Additionally, for legal reasons, could you take care of the CLA mentioned in our [contributing](https://github.com/GoogleCloudPlatform/DataflowJavaSDK/blob/master/CONTRIBUTING.md) file. This is a mandatory one-time process for every contributor who is not a Google employee.

Thanks.

**Comment by mxm** [comment]
Hi Davor! Thanks for accepting the pull request. I addressed the minor comments in the latest version of this pull request. I previously signed the CLA for another pull request to bdutil. So everything should be good to go.

**Comment by davorbonaci** [comment]
CLA confirmed. Pull request merged.

Thanks!

**Pull request merged**